docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Narayana LRA Participant Support
+
+include::./attributes.adoc[]
+
+== Introduction
+
+The LRA (short for Long Running Action) participant extension is useful in microservice based
+designs where different services can benefit from a relaxed notion of distributed consistency.
+
+The idea is for multiple services to perform different computations/actions in concert, whilst
+retaining the option to compensate for any actions performed during the computation.
+This kind of loose coupling of services bridges the gap between strong consistency models
+such as JTA/XA and "home grown" ad hoc consistency solutions.
+
+The model is based on the https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc#eclipse-microprofile-lra[Eclipse Microprofile LRA specification].
+The approach is for the developer to annotate a business method with a Java annotation
+(https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java[`@LRA`]).
+When such a method is called a context is created (if one is not already present) which is passed
+along with subsequent JAX-RS invocations until a method is reached
+which also contains an `@LRA` annotation with an attribute that indicates that the LRA should be
```

Comment:
I may have to look at the spec, but I'm confused between `@LRA` and LRA participants. I believe these terms need to be introduced.

Reply:
So you got your answer in the next paragraph. Do you think I need to do some reordering of the text where I introduce the various concepts?

Reply:
But I will include it here for clarity:

@LRA is the annotation for controlling the [life cycle of a Long Running Action](https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java#L185).

An LRA participant is a resource or POJO that participates in an LRA. It contains annotations for managing callbacks from the narayana LRA coordinator (via a proxy in the non JAX-RS case). Minimally, an LRA participant must contain a method annotated with @Compensate which is the callback to use when the LRA is cancelled and it is used to inform the participant that it should compensate for any actions it performed when the method marked with @LRA was invoked. This callback is guaranteed to be eventually delivered. And, optionally, if a method is annotated with @Complete then the participant will be notified if the LRA is closed.